product-service/src/services/db/queries.ts

```diff
@@ -0,0 +1,55 @@
+import { connect } from './connect';
+
+export const getProductsList = async () => {
+    const client = await connect();
+    try {
+        const { rows: products } = await client.query(`
+            select id, title, description, price, img_url, s.count 
+            from public.products p left join public.stocks s
+            on p.id = s.product_id
+        `);
+        return products;
+    } catch (error) {
+        throw error;
+    } finally {
+        client.end();
+    }
+};
+
+export const getProductById = async id => {
+    const client = await connect();
+    try {
+        const { rows: products } = await client.query(`
+            select id, title, description, price, img_url, s.count 
+            from public.products p left join public.stocks s
+            on p.id = s.product_id
+            where p.id = $1
+        `, [id]);
+        return products;
```

Comment:
it seems here it should be error 404, in case no product is found

Reply:
it is already in handler